subprojects/packagefiles/mpdecimal/meson.build

```diff
@@ -312,13 +365,15 @@ libmpdecpp_sources = files(
 libmpdecpp = library(
   'mpdec++',
   libmpdecpp_sources, mpdecimal_h,
-  dependencies : [m_dep, mpdec_dep],
-  version : library_version,
-  darwin_versions : [compat_version, library_version])
+  dependencies: [m_dep, mpdec_dep],
+  version: library_version,
+  soversion: library_soversion,
+  darwin_versions: [compat_version, library_version],
+  override_options: ['optimization=3'])
```

Comment:
Why is this optimization needed, instead of using whatever the user specified or setting it in default_options?

Reply:
The autotools build sets explicit and different optimization levels for the C and C++ libraries. I don't know if it is needed, but I thought the exercise is to replicate as close as possible what the native build rules do.

Reply:
I dunno. There may be an upstream reason that this specific optimization is needed, there may not be. Mainly I just wanted to make sure that this was being done for a reason -- the one you gave is fine.